internal/pkg/spire/spire.go

```diff
@@ -280,3 +282,42 @@ func formatIdUrl(id *types.SPIFFEID) (string, error) {
 		return id.String(), nil
 	}
 }
+
+// GetServerCABundleAndFederatedBundles retrieves the server CA bundle (i.e. bundle of the host) and any available
+// federated bundles from the SPIRE server, in order to do a federation health check
+func GetServerCABundleAndFederatedBundles(ctx context.Context, client *kube.Client) (string, map[string]string, error) {
+	serverCABundle, err := getServerCABundle(ctx, client)
+	if err != nil {
+		return "", nil, err
+	}
+	federatedBundles, err := getFederatedBundles(ctx, client)
+	if err != nil {
+		return "", nil, err
+	}
+	return serverCABundle, federatedBundles, err
+}
+
+func getServerCABundle(ctx context.Context, client *kube.Client) (string, error) {
+	command := []string{"bundle", "show"}
+	stdout, _, err := execInServerContainer(ctx, client, command)
+	return string(stdout), err
+}
+
+func getFederatedBundles(ctx context.Context, client *kube.Client) (map[string]string, error) {
+	command := []string{"bundle", "list"}
+	stdout, _, err := execInServerContainer(ctx, client, command)
+	return parseFederatedBundles(stdout), err
+}
+
+func parseFederatedBundles(stdout []byte) map[string]string {
+	bundles := make(map[string]string, 0)
+	split := strings.Split(string(stdout), "****************************************")
```

Comment:
This looks a bit brittle. It would be better to use `-output json`. The certificate is in a slightly different format, but the same format is available from `bundle show -output json` which would allow for comparison.

Reply:
Yeah `-output json` will be much better, totally overlooked that. Change incoming

Reply:
Moved over to the JSON method